x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pytorch/process/PyTorchResultProcessor.java

```diff
@@ -174,10 +192,10 @@ void processErrorResult(PyTorchResult result) {
 
         errorCount++;
 
-        logger.trace(() -> format("[%s] Parsed error with id [%s]", deploymentId, errorResult.requestId()));
```

Comment:
Seeing how `processResult` is synchronized and `processErrorResult` isn't, `errorCount` can be woefully incorrect due to race conditions.

That can be fixed in a different commit. This particular change looks good.

Reply:
Since it's quite a simple change I addressed it in https://github.com/elastic/elasticsearch/pull/88901/commits/1bdf3a0365005c8d65f597b847c986716fd52c45